packages/oruga/src/components/select/Select.vue

```diff
@@ -222,31 +222,41 @@ const emits = defineEmits<{
 const selectRef = ref<HTMLInputElement>();
 
 // use form input functionality
-const { checkHtml5Validity, onBlur, onFocus, onInvalid, setFocus } =
+const { checkHtml5Validity, onBlur, onFocus, onInvalid, setFocus, isValid } =
     useInputHandler(selectRef, emits, props);
 
 // inject parent field component if used inside one
 const { parentField, statusVariant, statusVariantIcon } = injectField();
 
-const vmodel = defineModel<T | T[]>({ default: undefined });
+// if id is given set as `for` property on o-field wrapper
+if (props.id) parentField?.value?.setInputId(props.id);
 
-const placeholderVisible = computed(() => !vmodel.value);
+const vmodel = defineModel<T | T[]>({
+    get: (v) => (isDefined(v) ? v : props.multiple ? [] : ""),
```

Comment:
I assume this affects the value emitted with the `update:modelValue` event, which might break some code that was expecting to get `null` values out. In particular, if the user is trying to use objects as the `modelValue`, they'd probably be quite surprised to get an empty string instead of `null` when the placeholder is selected. I'd say that's more a concern when dealing with the `options` prop (which provides a way to use object-valued options in a way that's actually compliant with Web standards) than it would be when using `<option>` elements (which are supposed to be strictly string-valued).

Reply:
Did I get the 'get' and 'set' wrong?
I wanted to set the internal representation of vmodel to `""` when it is `null` or `undefined` and pass `null` out of the select component when it is `""`.

Reply:
Based on the documentation, I'd expect `get` convert from the internal representation and `set` to convert _to_ the internal representation, so setting `v-model` to `""` would make the internal representation `null`, but the value exposed to events would be `""`. It sounds like you want the reverse, although that would break clients that do want set `v-model` to the empty string and expect to get an empty string back, which matches the standard behavior for a `<select>`.

I haven't tested any of this, though, so I could be wrong about the actual behavior.

Reply:
For instances that use the `options` prop, it might make sense to define the mapping based on that rather than forcing a mapping between `null` and strings here.

Reply:
I think I implemented it correctly. It is the same as the 'computed' implementation. 
I tested it and it worked.

Reply:
This does seem to work in practice, presumably because the "external" value of `v-model` is what gets passed to the `<select>`.

Reply:
As for breaking changes, because of how the placeholder actually works (not allowing selection of the placeholder after a "real" option has been selected), I haven't found any significant breaking changes in what I'd consider "reasonable" use. `<option :value="null">` is kind of a weird case; selecting the null-valued option actually causes the `<OSelect>` to emit its model value as `null`, which surprises me given how `vmodel` has been defined, but trying to set the model value to `null` from the parent component doesn't properly select the null-valued option. Using `<option :value="null">` is probably undesirable anyway, as it has the wrong semantics for constraint validation, so maybe that's not a major concern. The behavior does change for this example, though:

```vue
<script setup lang="ts">
import { OButton, OSelect } from '@oruga-ui/oruga-next';
import { ref, watch } from 'vue';

const val = ref<string | null>(null);

watch(val, v => console.debug(v));

</script>

<template>
  <main>
    <OField label="Value">
      <OSelect v-model="val">
        <option :value="null">null</option>
        <option value="1">1</option>
      </OSelect>
      <OButton :click="val = null">Set null</OButton>
    </OField>
  </main>
</template>
```